### PR TITLE
Rename serialized ForProvider as "forProvider"

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Manually tested using the example manifest provided in the repo, `examples/network/publicipaddress.yaml`.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/network/v1alpha3/types.go
Comment (ulucinar) [inline apis/network/v1alpha3/types.go:233]: Per Crossplane conventions, we had better rename the serialized form as `forProvider` instead of `properties`.

File: apis/network/v1alpha3/types.go
Comment (ulucinar) [inline apis/network/v1alpha3/types.go:281]: `spec.forProvider.sku` is optional. However, if a `spec.forProvider.sku` is specified, I think we need to have its name also specified. Thus removing the `omitempty` tag.

File: apis/network/v1alpha3/types.go
Comment (ulucinar) [inline apis/network/v1alpha3/types.go:308]: Per Crossplane conventions, let's move the observation under to `status.atProvider`.

File: apis/network/v1alpha3/types.go
Comment (ulucinar) [inline apis/network/v1alpha3/types.go:259]: Looks like the SDK constants are defined as `IPv4` and `IPv6`.

File: apis/network/v1alpha3/types.go
Comment (muvaf) [inline apis/network/v1alpha3/types.go:266]: Is this really optional?

File: apis/network/v1alpha3/types.go
Comment (muvaf) [inline apis/network/v1alpha3/types.go:236]: FYI, looks like there are [many more](https://docs.microsoft.com/tr-tr/rest/api/virtualnetwork/public-ip-addresses/create-or-update) properties that can be specified for `PublicIPAdress`.

File: apis/network/v1alpha3/types.go
Comment (ulucinar) [reply apis/network/v1alpha3/types.go]: Turns out not to be. Marked it as required.

File: apis/network/v1alpha3/types.go
Comment (ulucinar) [reply apis/network/v1alpha3/types.go:236]: We may consider adding them in later PRs unless there is immediate need, what do you think @muvaf?

Comment (muvaf) [reply apis/network/v1alpha3/types.go:236]: I personally prefer making it complete while I have the context of the resource, otherwise I'd have to load all that later again, but it can be merged as an alpha resource with less properties than full.

Comment (ulucinar) [reply apis/network/v1alpha3/types.go:267]: It turns out that we are not sure regarding the immediate requirements around this resource. Hence, I've extended the API surface.

File: pkg/controller/network/publicipaddress/managed.go
Comment (muvaf) [inline pkg/controller/network/publicipaddress/managed.go:108]: We usually update only if there is a diff and used to do `kube.Update` for late-init in old controllers but the recommended way today is to return `ResourceLateInitialized: true` to make managed reconciler handle it.

File: pkg/clients/network/network.go
Comment (muvaf) [inline pkg/clients/network/network.go:271]:
```suggestion
func checkDNSSettings(d *v1alpha3.PublicIPAddressDNSSettings, in *networkmgmt.PublicIPAddressDNSSettings) bool {
```

File: pkg/clients/network/network.go
Comment (muvaf) [inline pkg/clients/network/network.go:252]: `check*` sounds too generic IMO. Do you think we can use something like `is*UpToDate` as a name or something that tells what the function actually checks?

File: pkg/clients/network/network.go
Comment (muvaf) [inline pkg/clients/network/network.go:178]: nitpick: I usually go for a `Generate*Observation` and set the returned value to `cr.Status.AtProvider` in the controller logic so that readers don't have to come here to see what's set.

File: pkg/controller/network/publicipaddress/managed.go
Comment (muvaf) [inline pkg/controller/network/publicipaddress/managed.go:132]: Can you add a test for this function? I believe it'd be almost same as `Create` one.

**Review by muvaf — APPROVED** [review]
Thanks @ulucinar ! Looks good to merge after resolving non-nitpick comments.

File: pkg/controller/network/publicipaddress/managed.go
Comment (ulucinar) [reply pkg/controller/network/publicipaddress/managed.go:108]: Thanks @muvaf for the `ResourceLateInitialized: true` suggestion. I've removed the embedded controller-runtime client from our external client implementation and now always set `ObservationResult.ResourceLateInitialized` accordingly. 

We can at a later time revisit `provider-azure`'s late-initialization utility functions so that they report whether fields have been late-initialized. 

While working on this, I had also considered using `cmp.Equal` to detect any changes after late-initialization, however, I did not choose to do so as API server already detects noops and does not update the resource. However, leaving update calls to the managed reconciler is I think a good compromise.

Comment (ulucinar) [reply pkg/controller/network/publicipaddress/managed.go]: I just realized that if we leave this to the managed reconciler, we are losing any status updates. So, I'm reverting back to making the update call in-site as we already do in the repo:
https://github.com/crossplane/provider-azure/blob/8380dd2d85e0391bd02e23c5c47c13f8962c38a7/pkg/controller/database/postgresqlserver/managed.go#L122

So returning `ResourceLateInitialized: true` makes us lose status updates and I'd like to address proper calculation of this in a repo-wide manner.

**Pull request merged**
